scrabble.js

```diff
@@ -1,8 +1,146 @@
-var Scrabble = function() {};
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+// SCORING
+
+var Scoring = function() {
+  this.tiles = {
+    "A": 1,"E": 1,"I": 1,"O": 1,"U": 1,"L": 1,"N": 1,"R": 1,"S": 1,"T": 1,
+    "D": 2,"G": 2,"B": 3,"C": 3,"M": 3,"P": 3,
+    "F": 4,"H": 4,"V": 4,"W": 4,"Y": 4,"K": 5,"J": 8,"X": 8,"Q": 10,"Z": 10};
+  this.sevenLetterBonus = 50;
+};
+
+Scoring.prototype.score = function(word) {
+  // invalidWords.push(word); // global variable?
+  this.word = (word.toUpperCase());
+  var score = 0;
+  for (var i=0; i<this.word.length;i++) {
+    score += this.tiles[this.word[i]];
+  }
+  if (this.word.size <7) {
+      return score;
+  } else {
+      return score += this.sevenLetterBonus;
+  } //no check for words longer than seven
+};
+
+Scoring.prototype.highestScoreFrom = function(arrayOfWords) {
+  var maxScore = 0;
+  var maxWords = [];
+  var winningWord = 'PIZZAZZas';
+  this.words = arrayOfWords;
+
+  for (var i=0;i<this.words.length;i++) {
+    var currentScore = new Scoring().score(this.words[i]);
```

Comment:
Since this is inside of a loop, it will create a new instance every time you need to call the `score` function. In this case, I think it would make more sense to instantiate the object outside of the loop or switch the function to not rely on an individual instance.